### PR TITLE
feat: add nosklearnex option to run_examples

### PR DESCRIPTION
**Pull request by ahuber21**
Fixes the problem described in https://github.com/conda-forge/daal4py-feedstock/pull/57

### CODE REVIEW

File: tests/run_examples.py [hunk 1]
```diff
@@ -302,4 +298,29 @@ def run_all(nodist=False, nostream=False):
...
+        action="store_true",
+        default=False,
+        help="Skip examples with data streaming",
+    )
+    parser.add_argument(
+        "--nosklearnex",
+        action="store_true",
+        default=False,
+        help="Skip sklearnex examples",
+    )
```
Comment (samir-nasibli) [inline tests/run_examples.py:319]: When we are using this new argunment?

**Review by samir-nasibli — COMMENTED** [review]
Now it is better. Thank you!

File: tests/run_examples.py (hunk 1 above)
Comment (ahuber21) [reply tests/run_examples.py:319]: We should use it in conda recipe building to fix the problem discussed in the link from the description.

**Comment by napetrov** [comment]
Looks great - the even better one would be to add switch to run just skleranex components in case of scikit-learn-intelex recipe. 
In this case we don't need or even can running MB/daal4py examples and need just scikit-leanex. 

Might be this should be --sklernex  and --daal4py switches .

**Comment by ahuber21** [comment]
Adding `--sklearnex` and `--daal4py` switches to actively enable tests (with the default being disabled) would be incompatible with the current logic that has all tests enabled by default. But I added a `--nodaal4py` flag for symmetry.

File: tests/run_examples.py [hunk 2]
```diff
@@ -203,27 +196,31 @@ def get_exe_cmd(ex, nodist, nostream):
             return None
 
     if os.path.dirname(ex).endswith("daal4py"):
```
Comment (napetrov) [inline tests/run_examples.py:198]: i think we also have to have if for "mb" directory 
```suggestion
    if os.path.dirname(ex).endswith("daal4py") or os.path.dirname(ex).endswith("mb"):
```

Or might be add as separate if line

Comment (ahuber21) [reply tests/run_examples.py]: Good call!

**Review by napetrov — APPROVED** [review]

**Pull request merged**